src/NuGet.Core/NuGet.Configuration/Settings/AbstractSettingSection.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright(c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace NuGet.Configuration
+{
+    public sealed class AbstractSettingSection : SettingSection
```

Comment:
There are two types of sections... the ones that are actual sections that were read from a file and have a pointer to a settings file and the ones that are a collection of computer values... I decided to separate this two in `AbstractSettingSection` and `ParsedSettingSection` to have the code be a little easier to read, since the abstract sections are only an in-memory representation (i.e. don't map to one specific section in a file), but the parsed sections have a counterpart in a file.

Do you think this is not clear enough? what would you propose to make it better?

Reply:
@nkolev92 do you think `InMemorySettingSection` is a better name?